### PR TITLE
MagicLeap One ComputerVision FaceDetection standalone

### PR DESCRIPTION
**Pull request by lgarner**
This is the standalone version of the ComputerVision-GPU MagicLeap test app.

### CODE REVIEW

**Comment by brucearmstrong** [comment]
Do we still want the CompanionDevice and related code in this standalone version of the app?

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDevice.cs [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+      errorWriter.WriteLine("[" + TAG + "] " + msg);
+    }
+
+    // Debug Stdout:
+    [ConditionalAttribute("DEBUG")]
+    public static void D(string TAG, string msg)
+    {
+      Console.WriteLine("[" + TAG + "] " + msg);
+    }
+  }
```
Comment (ah1053) [inline unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDevice.cs:71]: I like this logging approach

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDevice.cs [hunk 2]
```diff
@@ -0,0 +1,479 @@
...
+        device_uuid = device.device_uuid == null ? "" : device.device_uuid,
+      };
+      /*
+      Logger.D("XXX", "RegisterDevice() has info.");
+
+      Logger.D("XXX", "Registering Device dn: " + ConfiguredDeviceRegisterRequest.user_name);
+      Logger.D("XXX", "Registering Device ue: " + ConfiguredDeviceRegisterRequest.user_email);
+      Logger.D("XXX", "Registering Device uid: " + ConfiguredDeviceRegisterRequest.user_id);
+      Logger.D("XXX", "Registering Device uuid: " + ConfiguredDeviceRegisterRequest.device_uuid);
+      */
```
Comment (ah1053) [inline unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDevice.cs:284]: Dead code ?

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDevice.cs [hunk 3]
```diff
@@ -0,0 +1,479 @@
...
+      Logger.D(TAG, "New location returning: {" + locData.Longitude + ", " + locData.Latitude + "}");
+      return new LocationReply
+      {
+        longitude = locData.Longitude,
+        latitude = locData.Latitude
+      };
+    }
+
+    // From CompanionApp.
+#if false
```
Comment (ah1053) [inline unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDevice.cs:359]: Is there a reason for not using (if())?

Comment (ah1053) [reply unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDevice.cs:359]: I guess this is for [ConditionalAttribute("DEBUG")] , correct ?

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/FaceDetection/FaceDetectionUI.cs [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+  }
+
+  // Base prefab for detection decoration.
+  public GameObject detectionPrefab;
+  private List<GameObject> detectionUIInsts;
+
+  public GameObject parentCam;
+
+  // public Camera cam = Camera.main;
+  ComputerVisionFaceDetection barcodeCamFloat;
```
Comment (ah1053) [inline unity/MagicLeapFaceDetectionPOC/Assets/Scripts/FaceDetection/FaceDetectionUI.cs:138]: Delete on 137
barcodeCamFloat is not used anywhere

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/FaceDetection/FaceDetectionUI.cs [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+
+}
+
+public class FaceDetectionUI : MonoBehaviour
+{
+  const string TAG = "FaceDetectionUI";
+  GameObject faceDetectionGo;
+
+  public delegate void DetectionCallback(int numDetected);
+  DetectionCallback aDetectionCallback;
```
Comment (ah1053) [inline unity/MagicLeapFaceDetectionPOC/Assets/Scripts/FaceDetection/FaceDetectionUI.cs:123]: Are you using aDetectionCallback somewhere?

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDeviceManager.cs [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+
+      return reply;
+    }
+
+    // TODO: Given there's no UI, the updater will just have to use the first one.
+    IEnumerator SelectAndRegisterDevice()
+    {
+      //UpdateDeviceList DevicecompanionDevice.DeviceList();
+      return null;
+    }
```
Comment (ah1053) [inline unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDeviceManager.cs:192]: I guess delete for now ?

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/FaceDetection/FaceDetectionUI.cs (hunk 5 above)
Comment (lgarner) [reply unity/MagicLeapFaceDetectionPOC/Assets/Scripts/FaceDetection/FaceDetectionUI.cs:123]: Nope, not in this code, removed.

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDevice.cs (hunk 3 above)
Comment (lgarner) [reply unity/MagicLeapFaceDetectionPOC/Assets/Scripts/CompanionDevice/CompanionDevice.cs]: Minor goal was keeping the general structure as this is just slimming down the dependencies of the original code.

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/FaceDetection/FaceDetectionUI.cs (hunk 4 above)
Comment (lgarner) [reply unity/MagicLeapFaceDetectionPOC/Assets/Scripts/FaceDetection/FaceDetectionUI.cs]: done.

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/ComputerVisionFaceDetection.cs [hunk 7]
```diff
@@ -0,0 +1,650 @@
...
+  GameObject cameraMonitor;
+  GameObject cameraSubtitle;
+  GameObject barcodeBuffer;
+  TextMeshPro tmpro;
+
+  // Some render targets to extract pixels for Camera.
+  RenderTexture tmp;
+  Texture2D nt;
+
+  private Color[] green;
```
Comment (ah1053) [inline unity/MagicLeapFaceDetectionPOC/Assets/Scripts/ComputerVisionFaceDetection.cs:80]: Couldn't find where it is used.

File: unity/MagicLeapFaceDetectionPOC/Assets/Scripts/ComputerVisionFaceDetection.cs [hunk 8]
```diff
@@ -0,0 +1,650 @@
...
+    }
+  }
+
+  string checkEdgeCloudletUri()
+  {
+    string uri = null;
+    if (CompanionDeviceManager.isReady())
+    {
+      uri = CompanionDeviceManager.GetAppInstHost();
+    }
```
Comment (ah1053) [inline unity/MagicLeapFaceDetectionPOC/Assets/Scripts/ComputerVisionFaceDetection.cs:215]: Couldn't find where this function is used.

**Review by brucearmstrong — APPROVED** [review]
Ship it.

**Review by ah1053 — APPROVED** [review]
⭐

**Review by franklin-huang-mobiledgex — APPROVED** [review]

**Pull request merged**